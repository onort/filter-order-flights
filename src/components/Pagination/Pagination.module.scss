@import "../../styles/main.scss";

.paginationContainer {
  display: flex;
  width: 1120px;
  justify-content: flex-end;
  margin: map-get($spacers, normal) 0;
}

.page {
  display: flex;
  width: px-to-rem(40px);
  height: px-to-rem(40px);
  align-items: center;
  justify-content: center;
  border: none;
  margin-left: map-get($spacers, small);
  background-color: $white;
  border-radius: px-to-rem(2px);
  box-shadow: 0 1px 5px rgba(0, 0, 0, 0.233752);
  color: $black;
  cursor: pointer;
  font-family: $font-family-primary;
  font-size: map-get($font-sizes, small);
  font-weight: 400;

  &.active {
    background-color: $primary;
    box-shadow: none;
    color: $white;
  }
}
