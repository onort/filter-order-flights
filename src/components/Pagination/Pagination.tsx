import React from "react"
import cx from "classnames"

import styles from "./Pagination.module.scss"
import { Container } from "../"

interface Props {
  activeIndex: number
  onClick: (page: number) => void
  totalPages: number
}

const Pagination: React.SFC<Props> = props => {
  const { activeIndex, totalPages } = props
  const pagesArray = Array.from(Array(totalPages).keys(), i => i + 1)

  return (
    <section className={styles.paginationWrapper}>
      <Container className={styles.paginationContainer}>
        {pagesArray.map(pageNumber => (
          <button
            key={pageNumber}
            className={cx(styles.page, {
              [styles.active]: activeIndex === pageNumber
            })}
            onClick={props.onClick.bind(null, pageNumber)}
          >
            {pageNumber}
          </button>
        ))}
      </Container>
    </section>
  )
}

export default Pagination
