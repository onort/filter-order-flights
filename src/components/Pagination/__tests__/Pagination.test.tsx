import React from "react"
import { shallow } from "enzyme"

import Pagination from "../"

const mockClickHandler = jest.fn()

describe("<Pagination />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <Pagination activeIndex={1} totalPages={2} onClick={mockClickHandler} />
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have correct amount of button elements", () => {
    const totalPages = 5
    const wrapper = shallow(
      <Pagination
        activeIndex={1}
        totalPages={totalPages}
        onClick={mockClickHandler}
      />
    )
    expect(wrapper.find("button")).toHaveLength(totalPages)
  })

  it("should have active class on correct child", () => {
    const wrapper = shallow(
      <Pagination activeIndex={2} totalPages={2} onClick={mockClickHandler} />
    )
    expect(wrapper.find(".active").text()).toBe("2")
  })

  it("should fire onClick when pageNumber is clicked", () => {
    const handleClick = jest.fn()
    const wrapper = shallow(
      <Pagination activeIndex={1} totalPages={3} onClick={handleClick} />
    )
    wrapper
      .find("button")
      .at(2)
      .simulate("click")
    expect(handleClick).toBeCalledTimes(1)
    expect(handleClick).toBeCalledWith(3)
  })
})
