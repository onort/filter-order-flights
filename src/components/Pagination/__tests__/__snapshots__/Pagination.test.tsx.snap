// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    activeIndex={1}
    onClick={[MockFunction]}
    totalPages={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Container
        className="paginationContainer"
      >
        <button
          className="page active"
          onClick={[Function]}
        >
          1
        </button>
        <button
          className="page"
          onClick={[Function]}
        >
          2
        </button>
      </Container>,
      "className": "paginationWrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <button
            className="page active"
            onClick={[Function]}
          >
            1
          </button>,
          <button
            className="page"
            onClick={[Function]}
          >
            2
          </button>,
        ],
        "className": "paginationContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": 1,
            "className": "page active",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 1,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": 2,
            "className": "page",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 2,
          "type": "button",
        },
      ],
      "type": [Function],
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Container
          className="paginationContainer"
        >
          <button
            className="page active"
            onClick={[Function]}
          >
            1
          </button>
          <button
            className="page"
            onClick={[Function]}
          >
            2
          </button>
        </Container>,
        "className": "paginationWrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <button
              className="page active"
              onClick={[Function]}
            >
              1
            </button>,
            <button
              className="page"
              onClick={[Function]}
            >
              2
            </button>,
          ],
          "className": "paginationContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": 1,
              "className": "page active",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 1,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": 2,
              "className": "page",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 2,
            "type": "button",
          },
        ],
        "type": [Function],
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Pagination /> should match snapshot 2`] = `"<section class=\\"paginationWrapper\\"><div class=\\"container paginationContainer\\"><button class=\\"page active\\">1</button><button class=\\"page\\">2</button></div></section>"`;
